src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -190,6 +196,11 @@ cpdef object compute_network(
     # columns: flow (qdc), velocity (velc), and depth (depthc) for each timestep
     # rows: indexed by data_idx
     cdef float[:,::1] flowveldepth = np.zeros((data_idx.shape[0], nsteps * 3), dtype='float32')
+    
+    # courant is a 2D float array that holds courant results
+    # columns: courant number (cn), kinematic celerity (ck), x parameter(X) for each timestep
+    # rows: indexed by data_idx
+    cdef float[:,::1] courant = np.zeros((data_idx.shape[0], nsteps * 3), dtype='float32')
```

Comment:
I'm not sure how to conditionally define variables. If `return_courant=False`, there is no need to define this very large array.

Reply:
I think you would need to do something like:
```
cdef float[:,::1] courant

if return_courant:
    courant = np.zeros((data_idx.shape[0], nsteps *3), dtype='float32')
```

Reply:
@awlostowski-noaa 
Did you try this logic to only set the array if the return will be calculated?